project3/coordinator/coordinator_resources/messenger_manager.cpp

```diff
@@ -33,7 +37,7 @@ bool MessengerManager::BroadcastMessage(MessageLog::Message msg) {
       // only log message once.
       // message will have timestamp of the first message sent.
       if (!is_logged) {
-        messenger->LogMessage(msg);
+        messenger->LogMessage(const_cast<MessageLog::Message*>(&msg));
```

Comment:
`const_cast` is really not something that should be used lightly. How about making `BroadcastMessage` take a pointer to `msg` instead? The way it is now, it tricks the caller into thinking that `msg` won't be modified, when in reality its timestamp will be updated.